### PR TITLE
#34: Don't return early when loading index, jus skip missing data

### PR DESCRIPTION
**Pull request by gwendalF**
The issue caused when deleting an entry is that the parsing failed. I propose to just skip line with missing data
https://github.com/ARK-Builders/arklib/issues/34

### CODE REVIEW

File: src/index.rs [hunk 1]
```diff
@@ -83,29 +82,62 @@ impl ResourceIndex {
...
-                    UNIX_EPOCH.add(Duration::from_millis(
-                        str.parse().map_err(|_| ArklibError::Parse)?,
-                    ))
+                    match parts.next() {
+                        Some(millis) => match millis.parse() {
+                            Ok(millis) => {
+                                UNIX_EPOCH.add(Duration::from_millis(millis))
+                            }
+                            Err(_) => {
+                                log::warn!("Can't parse {millis} to u64");
```
Comment (kirillt) [inline src/index.rs:97]: I think we should not continue the loop. If the file is broken, reading it makes no sense.

File: src/index.rs [hunk 2]
```diff
@@ -83,29 +82,62 @@ impl ResourceIndex {
...
+                                continue;
+                            }
+                        },
+                        None => {
+                            log::warn!("Missing timestamp from {entry}");
```
Comment (kirillt) [inline src/index.rs:102]: Same here
> I think we should not continue the loop. If the file is broken, reading it makes no sense.

File: src/index.rs [hunk 3]
```diff
@@ -83,29 +82,62 @@ impl ResourceIndex {
...
-                    let str = parts.next().ok_or(ArklibError::Parse)?;
-                    ResourceId::from_str(str)?
+                    match parts.next() {
+                        Some(str) => match ResourceId::from_str(str) {
+                            Ok(id) => id,
+                            Err(_) => {
+                                log::warn!(
+                                    "Can't parse {str} to a RessourceId"
+                                );
+                                continue;
```
Comment (kirillt) [inline src/index.rs:116]: Same here
> I think we should not continue the loop. If the file is broken, reading it makes no sense.

File: src/index.rs [hunk 4]
```diff
@@ -83,29 +82,62 @@ impl ResourceIndex {
...
+                            }
+                        },
+                        None => {
+                            log::warn!(
+                                "Missing ressource id part on the line {entry}"
+                            );
+                            continue;
```
Comment (kirillt) [inline src/index.rs:123]: Same here
> I think we should not continue the loop. If the file is broken, reading it makes no sense.

File: src/index.rs [hunk 5]
```diff
@@ -83,29 +82,62 @@ impl ResourceIndex {
...
-                index.insert_entry(path, IndexEntry { id, modified });
+                match CanonicalPathBuf::canonicalize(&path) {
+                    Ok(path) => {
+                        log::trace!("[load] {} -> {}", id, path.display());
+                        index.insert_entry(path, IndexEntry { id, modified });
+                    }
+                    Err(_) => {
+                        log::warn!("File {} not found", path.display());
+                        continue;
+                    }
```
Comment (kirillt) [inline src/index.rs:139]: Here continuing should be good :+1:

Comment (gwendalF) [reply src/index.rs:113]: Ok I've updated to return early if the file can't be parsed

**Review by kirillt — APPROVED** [review]

**Pull request merged**